src/Runner/Parallel/ProcessPool.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Runner\Parallel;
+
+use React\Socket\ServerInterface;
+
+/**
+ * Represents collection of active processes that are being run in parallel.
+ * Inspired by {@see https://github.com/phpstan/phpstan-src/blob/ed68345a82992775112acc2c2bd639d1bd3a1a02/src/Parallel/ProcessPool.php}.
+ *
+ * @author Greg Korba <greg@codito.dev>
+ *
+ * @internal
+ */
+final class ProcessPool
+{
+    private ServerInterface $server;
+
+    /** @var null|(callable(): void) */
+    private $onServerClose;
+
+    /** @var array<string, Process> */
+    private array $processes = [];
+
+    /**
+     * @param null|(callable(): void) $onServerClose
+     */
+    public function __construct(ServerInterface $server, ?callable $onServerClose = null)
+    {
+        $this->server = $server;
+        $this->onServerClose = $onServerClose;
+    }
+
+    public function getProcess(ProcessIdentifier $identifier): Process
+    {
+        if (!isset($this->processes[(string) $identifier])) {
+            throw ParallelisationException::forUnknownIdentifier($identifier);
+        }
+
+        return $this->processes[(string) $identifier];
+    }
+
+    public function addProcess(ProcessIdentifier $identifier, Process $process): void
+    {
+        $this->processes[(string) $identifier] = $process;
+    }
+
+    public function endProcessIfKnown(ProcessIdentifier $identifier): void
+    {
+        if (!isset($this->processes[(string) $identifier])) {
+            return;
+        }
+
+        $this->endProcess($identifier);
+    }
+
+    public function endAll(): void
+    {
+        foreach (array_keys($this->processes) as $identifier) {
+            $this->endProcess(ProcessIdentifier::fromRaw($identifier));
```

Comment:
question: why do we have 2 type of identifiers? "fixer-one" and "raw" ? can we use only fixer one" ?

Reply:
I don't get this concern... We have only one type of `ProcessIdentifier`, but some signatures require it as an object, and sometimes it comes as a raw string (e.g. from CLI input when worker is spawned) and object identifier has to be initialised.

Reply:
If I explicitly raise a "question" and asking for reason behind sth, consider to avoid resolving it on your owns, because it's me saying "I do not understand this" and you assuming that "now you do" (or "you don't need to" 😅 )

Reply:
I understand you want to have strong type of `ProcessIdentifier` instead of `string` 👍🏻 

Now I see that `fromRaw` behaves kind-of like an `unserialise` method, so we can recover object-type from previously autogenerated `php-cs-fixer_parallel_{ID}` value.